presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -174,6 +201,12 @@ public HashAggregationOperatorFactory(
             this.groupByTypes = ImmutableList.copyOf(groupByTypes);
             this.groupByChannels = ImmutableList.copyOf(groupByChannels);
             this.globalAggregationGroupIds = ImmutableList.copyOf(globalAggregationGroupIds);
+            this.sourceTypes = ImmutableList.copyOf(requireNonNull(sourceTypes, "orderByTypes is null"));
+            requireNonNull(orderByChannels, "orderByChannels is null");
+            checkArgument(orderByChannels.stream().map(List::size).mapToInt(Integer::intValue).sum() == 0 || ! spillEnabled, "ORDER BY in aggregation does not support spill");
```

Comment:
The `orderByChannels.stream().map(List::size).mapToInt(Integer::intValue).sum() == 0` expression could be extracted to a named method, otherwise it's hard to read.